collect_app/src/main/java/org/odk/collect/android/widgets/interfaces/DateTimeWidgetListener.java

```diff
@@ -0,0 +1,15 @@
+package org.odk.collect.android.widgets.interfaces;
+
+import android.content.Context;
+
+import org.javarosa.core.model.FormIndex;
+import org.joda.time.LocalDateTime;
+import org.odk.collect.android.logic.DatePickerDetails;
+
+public interface DateTimeWidgetListener {
```

Comment:
This class seems redundant you can use DateTimeWidgetUtils everywhere you need. Or maybe I missed something important?

Reply:
Oh actually I used to that I can verify the method calls to the `DateTimeWidgetUtils`, and then right separate tests to verify that the behaviour of the methods in `DateTimeWidgetUtils`. I can pass a mock of `DateTimeWidgetListener` in the Date time widgets constructor, and then verify method calls.

Reply:
But you don't need an interface to achieve that you can use mock with verify on `DateTimeWidgetUtils` in the same way. That's why I said it seems like a redundant abstraction.

Reply:
Yes right I can mock `DateTimeWidgetUtils` class and I was not able to use verify for any static method.

Reply:
But you don't verify static methods in your tests so I don't understand what the problem was. From what I see all tests will work without that interface too.

Reply:
The interface will act as a wrapper for the static methods, so I can verify any calls to them in the widget test, and I verified the behaviour of static method calls by verifying again the interface methods. 
In `DateTimeWidgetUtilsTest`, I used `dateTimeWidgetUtils.displayTimePickerDialog` to check if the Time Picker dialog is displayed whenever this method is called. This verifies the behaviour of static method `showTimePickerDialog`.

Reply:
Oh I understand, without that interface, I will have to check whether some dialog is displayed in the widget test, and I can verify whether correct arguments are passed, by checking the bundle of the displayed dialog fragment, then I can test separately whether the correct dialog class is displayed in the `DateTimeWidgetUtilsTest`. I'll remove the abstrcation.

Reply:
@grzesiek2010 I had some doubt regarding removing this abstraction completely. Currently, I'm verifying `dateTimeWidgetUtils.displayDatePickerDialog()` is called with valid arguments. I don't need to verify here that correct `datePickerDetails` argument is passed to the function, as that is extracted in the `DateTimeWidgetUtils` class. So I can write all the tests verifying that different dialogs are displayed for `datePickerType` only once for the tests for `DateTimeWidgetUtils` class. I need not verify this for all the widgets separately.

Reply:
Which test are you talking about? I did what I expected by myself to give it a try and all test you have added seem to work well https://github.com/getodk/collect/commit/b6e2850292f7b2cf9b2fe3a5226487b8d071b1a9

Reply:
Sorry for not understanding earlier. I understand now, I can simply remove the static methods as I cannot verify them, nonetheless, I access them using public wrapper functions, so I can simply verify them for the test.